app/models/user.rb

```diff
@@ -143,14 +143,6 @@ def send_verification_pin(app_name)
     TwilioService.new(self).sms_verification_pin(app_name)
   end
 
-  def channels
-    channels = Channel.private(self)
-    channels += Channel.reviewer if reviewer?
-    channels += Channel.supervisor if supervisor?
-    channels += Channel.order_fulfilment if order_fulfilment?
-    channels
-  end
```

Comment:
@abhinav098 This is another example of separation of concerns principle. I originally wrote this code a few years ago and incorrectly placed the channel detail inside the `User` object (because we needed access to user roles). There are no other uses of channel inside the User object so it is better and cleaner to extract them out into the `Channel` class  and call the publicly exposed `user.roleX?` methods from the `Channel` class.

This ultimately consolidates the logic for defining what channels a user subscribes too into just the `Channel` class instead of being spread out between `Channel` and `User` classes. It's a small change but makes it much easier to see what is going on now.